magicgui/_magicgui.py

```diff
@@ -23,13 +75,10 @@ def magicgui(  # noqa
     main_window: Literal[False] = False,
     app: AppRef = None,
     **param_options: dict,
-) -> FunctionGui:
-    ...
-
-
-@overload
+) -> FunctionGui[_R]: ...
```

Comment:
Pretty sweet use of typevars! =)

Reply:
I mean, I know this is exactly the point of them :joy:, I'd just never seen them in the wild. =)

Reply:
it _is_ lovely to work with! (now if I can only figure out how to move it all to `_magicgui.pyi`